src/prometheus_alert_rules/KubeflowMinioServices.rules

```diff
@@ -23,131 +23,61 @@ groups:
-    expr: minio_node_disk_free_bytes < 10 * 1024 * 1024 * 1024
-    for: 5m
-    labels:
-      severity: critical
-    annotations:
-      summary: "Low disk space on MinIO node"
-      description: |
-        MinIO node has less than 10GB of free disk space.
-        LABELS = {{ $labels }}
-
-  - alert: MinIO_High_Disk_Usage
-    expr: (minio_node_disk_used_bytes / minio_node_disk_total_bytes) > 0.9
+  # Disk space
+  - alert: Disk_Space_Filling_Up
+    expr: predict_linear(minio_node_disk_used_bytes[1h], 86400) >= minio_node_disk_total_bytes
     for: 5m
     labels:
       severity: high
     annotations:
-      summary: "High disk usage on MinIO node"
-      description: |
-        MinIO node has over 90% of its disk space used.
-        LABELS = {{ $labels }}
+      description: "MinIO will fill all available space in less than 1 day."
 
-- name: MinioS3RequestAlerts
-  rules:
-  - alert: MinIO_S3_Request_Errors
-    expr: rate(minio_s3_requests_errors_total[2m]) / rate(minio_s3_requests_total[2m]) > 0.1
-    for: 5m
-    labels:
-      severity: high
-    annotations:
-      summary: "High rate of S3 request errors"
-      description: |
-        More than 10% of S3 requests are resulting in errors.
-        LABELS = {{ $labels }}
-
-  - alert: MinIO_S3_Request_Auth_Failures
-    expr: rate(minio_s3_requests_rejected_auth_total[2m]) > 1
+  - alert: MinIO_Low_Disk_Space
+    expr: minio_node_disk_free_bytes < 1 * 1024 * 1024 * 1024
     for: 5m
     labels:
       severity: high
     annotations:
-      summary: "High rate of S3 authentication failures"
-      description: |
-        MinIO is experiencing more than 1 authentication failure per second in S3 requests.
-        LABELS = {{ $labels }}
+      description: "MinIO node has less than 1GB of free disk space."
 
-  - alert: MinIO_S3_Request_Invalid
-    expr: rate(minio_s3_requests_rejected_invalid_total[2m]) > 1
+  # File descriptors
+  - alert: MinIO_Too_Many_Open_File_Descriptors
+    expr: minio_node_file_descriptor_open_total >= minio_node_file_descriptor_limit_total
     for: 5m
     labels:
-      severity: high
+      severity: critical
     annotations:
-      summary: "Invalid S3 requests detected"
-      description: |
-        MinIO is rejecting more than 1 invalid S3 request per second.
-        LABELS = {{ $labels }}
+      description: "MinIO has opened the maximum number of file descriptors."
 
-  - alert: MinIO_S3_Requests_Waiting
-    expr: rate(minio_s3_requests_waiting_total[2m]) > 5
+  - alert: MinIO_Too_Many_Open_File_Descriptors_Rate
+    expr: minio_node_file_descriptor_open_total >= minio_node_file_descriptor_limit_total
     for: 5m
     labels:
       severity: high
     annotations:
-      summary: "S3 requests are waiting"
-      description: |
-        More than 5 S3 requests are persistently waiting per second.
-        LABELS = {{ $labels }}
+      description: "MinIO is on track to exhaust file descriptors in 24 hours."
```

Comment:
the expression and description here don't match, I think we should be using `predict_linear` similar to the first rule (Disk Space filling up)

Reply:
Ah, the problem here is that the expr is duplicated. Updated it in 7aa74ff